In [6]:
import pandas as pd

df = pd.read_csv('data/heart.csv')

categorical_cols = [
    'sex',
    'cp',
    'fbs',
    'restecg',
    'exang',
    'slope',
    'ca',
    'thal',
    'target'
]

for col in categorical_cols:
    
    print(f"\n===== {col} =====")
    
    freq_percent = df[col].value_counts(normalize=True) * 100
    
    print(freq_percent)


===== sex =====
sex
1    68.316832
0    31.683168
Name: proportion, dtype: float64

===== cp =====
cp
0    47.194719
2    28.712871
1    16.501650
3     7.590759
Name: proportion, dtype: float64

===== fbs =====
fbs
0    85.148515
1    14.851485
Name: proportion, dtype: float64

===== restecg =====
restecg
1    50.165017
0    48.514851
2     1.320132
Name: proportion, dtype: float64

===== exang =====
exang
0    67.326733
1    32.673267
Name: proportion, dtype: float64

===== slope =====
slope
2    46.864686
1    46.204620
0     6.930693
Name: proportion, dtype: float64

===== ca =====
ca
0    57.755776
1    21.452145
2    12.541254
3     6.600660
4     1.650165
Name: proportion, dtype: float64

===== thal =====
thal
2    54.785479
3    38.613861
1     5.940594
0     0.660066
Name: proportion, dtype: float64

===== target =====
target
1    54.455446
0    45.544554
Name: proportion, dtype: float64


task018
kaian


Sex -> درصد فراوانی مرد ها 68%و خانم ها 31% که نشون میده تعداد کرد بیشتر از زن هاست که یعنی  داده ها نسبتاً نا متوازن است ، پس میشه گفت این الگو ممکنه برای مردان عملکرد قوی تری داشته باشه تا زن ها 

Cp -> بیشتر افراد در دسته 0 قرار دارن با 47% و کمترین افراد در دسته 3 قرار دارن با 7% که نشون میده الگو برای افرادی که در دسته 0 هستن بهتر عمل کنه تا دسته 3 

fbs ->  خیلی نا متوازن است که این یعنی بیشتر دسته 0 در نظر گرفته شده و الگو پیدا شده پس ممکنه برای دسته 1 عملکرد درست انجام نشه

restecg -> می تونیم بگیم تقریبا متعادله و عملکرد برای هر دو به یک اندازه خوبه 

exang -> همونطور که مشخصه داده ها بیشتر در دسته 0 هستن و و این یعنی متعادل نیست و مدل الگو مربوط به این دسته رو بهتر یاد میگیره و عملکردش برای افراد در دسته 1 نسبتاً ضعیف تره 

Slop -> درصد فراوانی در دسته 1,2 تقریبا یکسانه اما دسته 0 درصد فراوانی خیلی کمی دارد که یعنی ممکنه برای افراد مربوط به این دسته به درستی الگو پیدا نکند

ca -> توزیع داده به صورت نا متوازن است که دسته 4,3 کمترین فراوانی رو دارن و ممکنه مدل نتونه برای افرادی که در این دسته هستند الگو مناسبی پیدا کنه 

thal-> توزیع به صورت نسبتاً نا متوازن است و دسته 2,3 دارای بیشترین درصد فراوانی هستند ولی دسته 1 درصد کمی داره ممکنه مدل نتونه الگو رو درست یاد بگیره و برای دسته آخر که تقریبا 0 است می توانیم بگوییم که مدل چیزی از این دسته یاد نمیگیرد  

target -> نسبتاً متعادله و درصد داده های دو دسته خیلی بهم نزدیک هستن پس برای هر دو دسته داده کافی داره